x-pack/filebeat/input/lumberjack/server_test.go

```diff
@@ -0,0 +1,238 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package lumberjack
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/transport/tlscommon"
+	client "github.com/elastic/go-lumber/client/v2"
+)
+
+const testTimeout = 10 * time.Second
+
+func TestServer(t *testing.T) {
+	makeTestConfig := func() config {
+		var c config
+		c.InitDefaults()
+		c.ListenAddress = "localhost:0"
+		c.MaxConnections = 1
+		c.Keepalive = time.Second
+		c.Timeout = time.Second
+		return c
+	}
+
+	t.Run("empty_batch", func(t *testing.T) {
+		testSendReceive(t, makeTestConfig(), 0, nil)
+	})
+
+	t.Run("no tls", func(t *testing.T) {
+		testSendReceive(t, makeTestConfig(), 10, nil)
+	})
+
+	t.Run("tls", func(t *testing.T) {
+		clientConf, serverConf := tlsSetup(t)
+		clientConf.Certificates = nil
+
+		c := makeTestConfig()
+		c.TLS = serverConf
+		c.TLS.ClientAuth = 0 // tls.NoClientCert
+		c.TLS.VerificationMode = tlscommon.VerifyNone
+
+		testSendReceive(t, c, 10, clientConf)
+	})
+
+	t.Run("mutual tls", func(t *testing.T) {
+		clientConf, serverConf := tlsSetup(t)
+
+		c := makeTestConfig()
+		c.TLS = serverConf
+
+		testSendReceive(t, c, 10, clientConf)
+	})
+}
+
+func testSendReceive(t testing.TB, c config, numberOfEvents int, clientTLSConfig *tls.Config) {
+	require.NoError(t, logp.TestingSetup())
+	log := logp.NewLogger(inputName).With("test_name", t.Name())
+
+	ctx, shutdown := context.WithTimeout(context.Background(), testTimeout)
+	t.Cleanup(shutdown)
```

Comment:
The `go test` command already accepts a timeout parameter, do you really need to have it hard coded and manually set?